cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/KafkaClusterState.java

```diff
@@ -191,18 +224,21 @@ private void populateKafkaBrokerState(Map<Integer, Integer> leaderCountByBrokerI
     Set<PartitionInfo> offlinePartitions = new HashSet<>();
     Set<PartitionInfo> otherPartitions = new HashSet<>();
     Set<PartitionInfo> partitionsWithOfflineReplicas = new HashSet<>();
+    Set<PartitionInfo> underMinIsrPartitions = new HashSet<>();
 
     populateKafkaPartitionState(underReplicatedPartitions,
                                 offlinePartitions,
                                 otherPartitions,
                                 partitionsWithOfflineReplicas,
+                                underMinIsrPartitions,
                                 verbose);
 
     // Write the partition state.
     Map<String, List> kafkaClusterByPartitionState = new HashMap<>();
     kafkaClusterByPartitionState.put(OFFLINE, getJsonPartitions(offlinePartitions));
     kafkaClusterByPartitionState.put(WITH_OFFLINE_REPLICAS, getJsonPartitions(partitionsWithOfflineReplicas));
     kafkaClusterByPartitionState.put(URP, getJsonPartitions(underReplicatedPartitions));
+    kafkaClusterByPartitionState.put(UNDER_MIN_ISR, getJsonPartitions(underMinIsrPartitions));
```

Comment:
Based on our offline discussion with Kun:
As Kun pointed out, there could be partitions that are not `URP`, but are still `under min isr`, because they have a replication factor less than the `minIsr` configuration (i.e. misconfiguration). We should also list them here. For ease of extracting the information, we will list the underminISR list as a distinct list.